### DISCUSSION TITLE
FLV support for rendering video and generating waveform

### DISCUSSION THREAD

**Discussion by faisalpathan** [katspaugh/wavesurfer.js#2874] [Q&A] [answered]
Hey, I had implemented video annotation by loading the video element, but it supports standard video formats like MP4, WebM, and OGG.

I need to render a flv video but since it's not supported by the native video player, do we have any other way to render it and show waveforms in wavesurfer.

Research:

Through research, I got to know that flv can work with video.js, and we need to integrate video.js with wavesurfer. But I am not sure if that will solve the problem of generating waveform.

[videojs-wavesurfer](https://github.com/collab-project/videojs-wavesurfer) 
 
Wavesurfer.js version(s):
[6.6.3](https://github.com/katspaugh/wavesurfer.js/releases/tag/6.6.3)

Browser and operating system version(s):
Browser - Chrome v113.0.5672.126
OS - MacOS Ventura

**Comment by katspaugh** [comment, marked as answer]
There might be a way to decode flv files with a custom decoder, either in the browser or outside of the browser, and use pre-decoded peaks with wavesurfer.

**Reply by faisalpathan** [reply]
@katspaugh, thanks for the response, does wavesurfer take a custom decoder in params or use pre-decoded peaks the best way possible?

**Reply by katspaugh** [reply]
Pre-decoded peaks are preferred. `wavesurfer.load(audioUrl, peaks, duration)`. Duration is optional.

**Comment by faisalpathan** [comment]
Thanks @katspaugh
